homeassistant/components/ezviz/image.py

```diff
@@ -51,12 +55,28 @@ def __init__(
         self._attr_image_last_updated = dt_util.parse_datetime(
             str(self.data["last_alarm_time"])
         )
+        camera = hass.config_entries.async_entry_for_domain_unique_id(DOMAIN, serial)
+        self.alarm_image_password = (
+            camera.data[CONF_PASSWORD] if camera is not None else ""
+        )
 
     async def _async_load_image_from_url(self, url: str) -> Image | None:
         """Load an image by url."""
         if response := await self._fetch_url(url):
+            image_data = response.content
+            if self.data["encrypted"] and self.alarm_image_password:
+                try:
+                    image_data = decrypt_image(
+                        response.content, self.alarm_image_password
+                    )
+                except PyEzvizError:
```

Comment:
```suggestion
            if TYPE_CHECKING:
                assert self.platform.config_entry
            alarm_image_password = self.platform.config_entry.data[CONF_PASSWORD]
            image_data = response.content
            if self.data["encrypted"] and alarm_image_password:
                try:
                    image_data = decrypt_image(
                        response.content, alarm_image_password
                    )
                except PyEzvizError:
```

Reply:
Last Motion image is loaded from Ezviz Cloud. `config_entry` here contains data for Ezviz Cloud account, like `url`/`session`. It doesn't contain camera password, stored in another config.
How we can get `config_entry` for camera by its `serial` here?

Reply:
Thanks @DeKaN for clarifying